### ISSUE TITLE
Compile error under Windows using MSYS2

### ISSUE THREAD

**Issue by graphixillusion** [Cpasjuste/pemu#38]
Hi there. I've followed the guide for building under Windows with MSYS2. I got an error around 4% of building process. The error is:

```
D:/testlab/pemu/libcross2d/source/platforms/gl2/gl_renderer.cpp: In member function 'void c2d::GLRenderer::initGL()':
D:/testlab/pemu/libcross2d/source/platforms/gl2/gl_renderer.cpp:26:5: error: 'gladLoadGL' was not declared in this scope
   26 |     gladLoadGL();
      |     ^~~~~~~~~~
make[2]: *** [libcross2d/CMakeFiles/cross2d.dir/build.make:574: libcross2d/CMakeFiles/cross2d.dir/source/platforms/gl2/gl_renderer.cpp.obj] Error 1
make[2]: exit from directory «/d/testlab/pemu/cmake-build»
make[1]: *** [CMakeFiles/Makefile2:259: libcross2d/CMakeFiles/cross2d.dir/all] Error 2
make[1]: exit from directory «/d/testlab/pemu/cmake-build»
make: *** [Makefile:147: all] Error 2
```

**Comment by Cpasjuste** [comment]
Hi, 

This should be fixed in latest updates ?

**Comment by graphixillusion** [comment]
@Cpasjuste i got this error now:

```
[  7%] Linking CXX executable cross2d_test.exe
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/10.1.0/../../../../x86_64-w64-mingw32/bin/ld.exe: C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/10.1.0/../../../../lib\libfreetype.a(sfnt.o):(.text+0xae3f): undefined reference to `BrotliDecoderDecompress'
collect2.exe: error: ld returned 1 exit status
make[2]: *** [libcross2d/CMakeFiles/cross2d_test.dir/build.make:107: libcross2d/cross2d_test.exe] Error 1
make[1]: *** [CMakeFiles/Makefile2:235: libcross2d/CMakeFiles/cross2d_test.dir/all] Error 2
make: *** [Makefile:150: all] Error 2
```

**Comment by jjslshn** [comment]
i hope to build it, some  hack game need to modify the CRC32 code. but it is a difficulty for me .
msys2 and vitasdk ,and followed the guide for building ,will appear  kinds of incorrect。

**Comment by salamcast** [comment]
I'm having problems building this under windows as well.  it seems to fail with references to zip.exe 

example: 
`/bin/sh: line 1: C:msys64usrbinzip.exe: command not found`

but in the  `pnes/CMakeFiles/pnes.data.extra.dir/build.make` file it was written as `C:\msys64\usr\bin\zip.exe`

the error went away when i changed it to `/C/msys64/usr/bin/zip.exe.`

I still got other build stopping error after the replacement of the reference to zip.exe in the above file, its not only the pnes build.make file, but also the psnes and pfba that also have the wrong/unrecognized reference for zip.exe.

```
[  8%] Building CXX object pnes/CMakeFiles/pnes.dir/__/cores/nestopia/source/common/cli.cpp.obj
In file included from C:/msys64/home/dude/pemu/cores/nestopia/source/common/cli.cpp:28:
C:/msys64/home/dude/pemu/cores/nestopia/source/common/config.h:74:12: warning: 'int config_match(void*, const char*, const char*, const char*)' declared 'static' but never defined [-Wunused-function]
   74 | static int config_match(void* user, const char* section, const char* name, const char* value);
      |            ^~~~~~~~~~~~
cc1plus.exe: note: unrecognized command-line option '-Wno-unused-definition' may have been intended to silence earlier diagnostics
[  8%] Building CXX object pnes/CMakeFiles/pnes.dir/__/cores/nestopia/source/common/homebrew.cpp.obj
In file included from C:/msys64/home/dude/pemu/cores/nestopia/source/common/homebrew.cpp:23:
C:/msys64/home/dude/pemu/cores/nestopia/source/common/config.h:74:12: warning: 'int config_match(void*, const char*, const char*, const char*)' declared 'static' but never defined [-Wunused-function]
   74 | static int config_match(void* user, const char* section, const char* name, const char* value);
      |            ^~~~~~~~~~~~
cc1plus.exe: note: unrecognized command-line option '-Wno-unused-definition' may have been intended to silence earlier diagnostics
[  8%] Building CXX object pnes/CMakeFiles/pnes.dir/__/cores/nestopia/source/common/ini.cpp.obj
In function 'char* strncpy0(char*, const char*, size_t)',
    inlined from 'int ini_parse_file(FILE*, int (*)(void*, const char*, const char*, const char*), void*)' at C:/msys64/home/dude/pemu/cores/nestopia/source/common/ini.cpp:164:25:
C:/msys64/home/dude/pemu/cores/nestopia/source/common/ini.cpp:76:12: warning: 'char* strncpy(char*, const char*, size_t)' output may be truncated copying 49 bytes from a string of length 199 [-Wstringop-truncation]
   76 |     strncpy(dest, src, size);
      |     ~~~~~~~^~~~~~~~~~~~~~~~~
At global scope:
cc1plus.exe: note: unrecognized command-line option '-Wno-unused-definition' may have been intended to silence earlier diagnostics
[  8%] Building CXX object pnes/CMakeFiles/pnes.dir/__/cores/nestopia/source/common/input.cpp.obj
In file included from C:/msys64/home/dude/pemu/cores/nestopia/source/common/input.cpp:26:
C:/msys64/home/dude/pemu/cores/nestopia/source/common/config.h:74:12: warning: 'int config_match(void*, const char*, const char*, const char*)' declared 'static' but never defined [-Wunused-function]
   74 | static int config_match(void* user, const char* section, const char* name, const char* value);
      |            ^~~~~~~~~~~~
cc1plus.exe: note: unrecognized command-line option '-Wno-unused-definition' may have been intended to silence earlier diagnostics
[  8%] Building CXX object pnes/CMakeFiles/pnes.dir/__/cores/nestopia/source/common/nstcommon.cpp.obj
In file included from C:/msys64/mingw64/include/zconf.h:475,
                 from C:/msys64/mingw64/include/zlib.h:34,
                 from C:/msys64/home/dude/pemu/ui/minizip/unzip.h:51,
                 from C:/msys64/home/dude/pemu/cores/nestopia/source/common/nstcommon.cpp:47:
C:/msys64/mingw64/x86_64-w64-mingw32/include/unistd.h: In function 'int ftruncate(int, off32_t)':
C:/msys64/mingw64/x86_64-w64-mingw32/include/unistd.h:67:10: error: '_chsize' was not declared in this scope; did you mean '_msize'?
   67 |   return _chsize (__fd, __length);
      |          ^~~~~~~
      |          _msize
In file included from C:/msys64/home/dude/pemu/cores/nestopia/source/common/nstcommon.cpp:52:
C:/msys64/home/dude/pemu/cores/nestopia/source/common/config.h: At global scope:
C:/msys64/home/dude/pemu/cores/nestopia/source/common/config.h:74:12: warning: 'int config_match(void*, const char*, const char*, const char*)' declared 'static' but never defined [-Wunused-function]
   74 | static int config_match(void* user, const char* section, const char* name, const char* value);
      |            ^~~~~~~~~~~~
cc1plus.exe: note: unrecognized command-line option '-Wno-unused-definition' may have been intended to silence earlier diagnostics
make[2]: *** [pnes/CMakeFiles/pnes.dir/build.make:225: pnes/CMakeFiles/pnes.dir/__/cores/nestopia/source/common/nstcommon.cpp.obj] Error 1
make[1]: *** [CMakeFiles/Makefile2:530: pnes/CMakeFiles/pnes.dir/all] Error 2
make: *** [Makefile:149: all] Error 2

```


Hope this is useful, maybe you can post a windows binary soon

**Comment by Cpasjuste** [comment]
Yep, sorry, from memory there was a lot of hackering to get it compile on windows. I may take a look someday, but in the meantime you may have more luck with ubuntu for windows 10.

**Comment by Cpasjuste** [comment]
Hi, should be fixed in [6.2](https://github.com/Cpasjuste/pemu/releases/tag/v6.2) (currently building, files will be available soon). If not, be sure to report here and i'll re-open the issue.
